xds/src/main/java/io/grpc/xds/internal/sds/SdsSslContextProviderHelper.java

```diff
@@ -178,9 +104,9 @@ public void addCallback(Callback callback, Executor executor) {
   }
 
   private void callPerformCallback(
-      Callback callback, Executor executor, final SslContext sslContextCopy) {
-    performCallback(
-        new SslContextGetter() {
+      SslContextProvider.Callback callback, Executor executor, final SslContext sslContextCopy) {
+    sslContextProvider.performCallback(
```

Comment:
helper using helpee is weird. is there way to avoid this circular dependency? btw, this class makes sense as an abstract class, but not as a helper class.

Reply:
Yes I realized that but Java doesn't have multiple inheritance (unlike C++) so my only alternative was to inject the helper to achieve code reuse - I really would have hated code duplication.

I was also aware of the Helper class calling back into the caller of the helper but again I thought this is the best way to structure the code without unnecessarily duplicating it. Let me see if I can achieve the same result by restructuring this.